### PR TITLE
Add model fitting tab docs and update muon docs

### LINKED ISSUE #33689: Muon docs
Want to add muon docs for:

- [x] feature flags
- [ ] model analysis
- [ ] raw plots

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**

This PR adds documentation to the muon Model Fitting tab as well as updating other tabs to include correct refferences to it,

**To test:**

build and check docs for propper spelling and grammer

Fixes #33689

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/interfaces/muon/muon_analysis_plotting_window.rst [hunk 1]
```diff
@@ -19,13 +19,6 @@ It displays the selected data for a fit and if a fit is requested this is also s
 In the sequential fitting tab, the user can select the data to plot by highlighting the different rows in the table.
 
 
-Model Data
```
Comment (AnthonyLim23) [inline docs/source/interfaces/muon/muon_analysis_plotting_window.rst:22]: why have we deleted this?

Comment (Harrietbrown) [reply docs/source/interfaces/muon/muon_analysis_plotting_window.rst:22]: The fit data window is only in Muon analysis so they are sitll different.

Comment (AnthonyLim23) [reply docs/source/interfaces/muon/muon_analysis_plotting_window.rst:22]: ah yes good spot!

File: docs/source/interfaces/muon/muon_model_fitting_tab.rst [hunk 2]
```diff
@@ -7,9 +7,15 @@ Model Fitting Tab
...
-The selected model fit data is displayed in a plotting pane on the right hand side of the interface. An example model fit plot is seen in the image below.
+Overview
+^^^^^^^^
+
+The model fitting tab is used to perform a fit across fit parameters and sample logs found in a results table within the
+:ref:`Muon Analysis <Muon_Analysis-ref>` and :ref:`Frequency Domain Analysis <Frequency_Domain_Analysis-ref>`. To use
+Model fitting it must first be enabled via the :ref:`feature flags <Muon_Feature_Flags-ref>`.
+This tab allows you to select a results table, before selecting the data that should be plotted on the x and y axes. It
+is then possible to perform a fit using a selected fit function. The selected model fit data is displayed in a plotting
+pane on the right hand side of the interface. An example model fit plot is seen in the image below.
```
Comment (AnthonyLim23) [inline docs/source/interfaces/muon/muon_model_fitting_tab.rst:18]: need to add FDA to the used by list:

:ref:`Frequency Domain Analysis <Frequency_Domain_Analysis-ref>`

Comment (Harrietbrown) [reply docs/source/interfaces/muon/muon_model_fitting_tab.rst:18]: It already says that it is used by FDA

Comment (AnthonyLim23) [reply docs/source/interfaces/muon/muon_model_fitting_tab.rst:18]: Sorry its a bit confusing. I meant at the end of the file. I couldnt comment there as it wasn't edited.

**Review by AnthonyLim23 — APPROVED** [review]
looks good

**Pull request merged**